### ISSUE TITLE
[java] UR anomaly false positive? 

### ISSUE THREAD

**Issue by davidgoate** [pmd/pmd#1120]
**Affects PMD Version:** 6.1.0

**Rule:** UR - Anomaly

**Description:** 

>There is a reference to a variable that was not defined before. This is a bug and leads to an error. 

I don't think this is the case?

**Code Sample demonstrating the issue:**

Complains about use of variable `result`.

```java

import org.apache.commons.csv.CSVFormat;
import org.apache.commons.csv.CSVPrinter;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.util.Collection;

public class Dummy {

    public String dummy(final CSVRecord header, final Collection<CSVRecord> rows) {
        final StringBuilder result = new StringBuilder();
        try (CSVPrinter csvWriter = new CSVPrinter(result, "")) {
            csvWriter.printRecord(header);
            csvWriter.printRecords(rows);
            csvWriter.flush();
            return result.toString();
        } catch (final IOException ioe) {
            throw new RuntimeException("Unable to write CSV", ioe);
        }
    }
}
```

**Running PMD through:** *CLI*

**Comment by jsotuyod** [comment]
I'm merging this onto #1636, as we plan to drop UR-anomaly detection
